### ISSUE TITLE
Support file-based configuration

### ISSUE THREAD

**Issue by sam-mccall** [google/mt#3]
Your opinions on fonts and colors are incorrect :-)

Happy to take a stab at this, but wanted your opinion before writing code. Proposal:

  - the current stuff in `config.h` becomes a `struct config`, char[] becomes string etc.
  - existing values become the defaults
  - defaults can be overridden using $HOME/.mt (not configurable, not via flags, etc)
  - format is key=value, e.g
```
    # This is a comment
    font=Roboto Mono for Powerline:size=9:antialias=true:autohint=true
    colors[8]=#cccccc
    keys[shift+KP_Home]=\033[1;2H
```
  - config parser is ad-hoc and initially only handles most important options (font, colors, window size). Keys are deferred to later because of complexity.
  - missing .mt is ignored, unreadable file prints a warning, unparseable lines print a warning each

**Comment by chandlerc** [comment]
Before we do this, the config.h needs to be cleaned up imo. A bunch of the stuff there should move to the source code.

I'd also like to see if we can have a common but of infra to handle flags and a config file.... Lemme put some thought into that...

**Comment by sam-mccall** [comment]
> the config.h needs to be cleaned up

Agree, though no need to do this *first* I think - just don't make the config parser support changing them?
In particular, we can allow font selection without first deciding whether forceselmod should be configurable or not :)

But happy to move some of the obvious stuff inline.

> I'd also like to see if we can have a common but of infra to handle flags and a config file.... Lemme put some thought into that...

Sure. Obvious approaches are:

 - flag `--key=value` is equivalent to `key=value` in the config file, and takes precedence
 - use real libraries for flags/config, ideally one that relates them together

but I'm curious about less-obvious ones!

**Comment by sam-mccall** [comment]
First step in cleaning up config.h is understanding what everything there does!
#4 is my first step in that direction...

**Comment by eddelbuettel** [comment]
That would be nice.  I built it twice on two different Ubuntu 16.10 machines and had inconsistent results with fonts. I quite like the idea of mt as a tmux/byobu user -- but it has to look _crisp_ :)
